XamlAnimatedGif/Animator.cs

```diff
@@ -43,6 +46,43 @@ internal Animator(Stream sourceStream, Uri sourceUri, GifDataStream metadata, Re
             _previousBackBuffer = new byte[desc.Height * _stride];
             _indexStreamBuffer = CreateIndexStreamBuffer(metadata, _sourceStream);
             _timingManager = CreateTimingManager(metadata, repeatBehavior);
+
+            _cacheFrameDataInMemory = cacheFrameDataInMemory;
+
+            if (!_cacheFrameDataInMemory)
+                return;
+
+            var frameDescs = new GifImageDescriptor[_metadata.Frames.Count];
+            var frameRows = new IEnumerable<int>[_metadata.Frames.Count];
+            for (var frameIndex = 0; frameIndex < _metadata.Frames.Count; frameIndex++)
+            {
+                var currentFrame = _metadata.Frames[frameIndex];
+                var frameDesc = currentFrame.Descriptor;
+                var currentRect = GetFixedUpFrameRect(frameDesc);
+                var currentFrameRows = frameDesc.Interlace
+                    ? InterlacedRows(currentRect.Height)
+                    : NormalRows(currentRect.Height);
+                
+                frameDescs[frameIndex] = frameDesc;
+                frameRows[frameIndex] = currentFrameRows.ToArray();
+            }
+            _loadFramesDataTask = Task.Run(() =>
+            {
+                for (var frameIndex = 0; frameIndex < _metadata.Frames.Count; frameIndex++)
+                {
+                    var frame = _metadata.Frames[frameIndex];
+                    var frameDesc = frameDescs[frameIndex];
+                    var rows = frameRows[frameIndex];
+                    var indexCompressedBytes = GetIndexBytesAsync(frameIndex, CancellationToken.None).Result;
+                    using var indexDecompressedStream = new LzwDecompressStream(indexCompressedBytes, frame.ImageData.LzwMinimumCodeSize);
+                    _cachedFrameBytes[frameIndex] = rows.Select((r, i) =>
```

Comment:
You never use the values of `r` and `i`.
I understand why, but it means that you don't actually need `rows`, or worry about whether the rows are interlaced or not (this is handled during rendering in `RenderFrameAsync`) You could just use `Enumerable.Range(0, height)` instead of `rows`, for the same result.

Reply:
I also converted it to a simple for block and now the code is much simpler